<a href="https://colab.research.google.com/github/FunnyCorner/PROJECT-2.-HeadHunter/blob/main/Project2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [105]:
import psycopg2
import pandas as pd

# Параметры подключения
DBNAME = 'project_sql'
USER = 'skillfactory'
PASSWORD = 'cCkxxLVrDE8EbvjueeMedPKt'
HOST = '84.201.134.129'
PORT = 5432

# Подключение к базе данных
connection = psycopg2.connect(
    dbname=DBNAME,
    user=USER,
    host=HOST,
    password=PASSWORD,
    port=PORT
)

# Создание курсора
cursor = connection.cursor()


3.1Сколько вакансий есть в базе?

In [50]:

query = "SELECT COUNT(*) as vacancy_count FROM vacancies;"
cursor.execute(query)

# Извлечение результата
vacancy_count = cursor.fetchone()[0]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Количество вакансий в базе: {vacancy_count}")

Количество вакансий в базе: 49197


3.2Сколько работодателей?

In [35]:
query = "SELECT COUNT(*) as employer_count FROM employers;"
cursor.execute(query)

# Извлечение результата
employer_count = cursor.fetchone()[0]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Количество работодателей в базе: {employer_count}")

Количество работодателей в базе: 23501


3.3Сколько регионов?



In [38]:
query = "SELECT COUNT(*) as area_count FROM areas;"
cursor.execute(query)

# Извлечение результата
area_count = cursor.fetchone()[0]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Количество регионов в базе: {area_count}")

Количество регионов в базе: 1362


3.4Сколько сфер деятельности?

In [40]:
query = "SELECT COUNT(*) as industry_count FROM industries;"
cursor.execute(query)

# Извлечение результата
industry_count = cursor.fetchone()[0]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Количество сфер деятельности в базе: {industry_count}")

Количество сфер деятельности в базе: 294


Выводы. Задания 3.1 по 3.4:В базе данных содержится 49197 вакансий от 23501 работодателей, что свидетельствует о значительном количестве предложений на рынке труда.
Вакансии распределены по 1362 регионам и охватывают 294 сферы деятельности, что указывает на широкий географический и профессиональный охват.

4.1Напишите запрос, который позволит узнать, сколько (cnt) вакансий в каждом регионе (area).

Выберите пятёрку лидеров по количеству вакансий:

In [47]:
query = """
    SELECT
        a.name as area,
        COUNT(v.id) as cnt
    FROM
        vacancies v
    JOIN
        areas a ON v.area_id = a.id
    GROUP BY
        a.name
    ORDER BY
        cnt DESC
    LIMIT 5;
"""
cursor.execute(query)

# Извлечение результата
top_areas = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результатов
for area, cnt in top_areas:
    print(f"Регион: {area}, Количество вакансий: {cnt}")

Регион: Москва, Количество вакансий: 5333
Регион: Санкт-Петербург, Количество вакансий: 2851
Регион: Минск, Количество вакансий: 2112
Регион: Новосибирск, Количество вакансий: 2006
Регион: Алматы, Количество вакансий: 1892


4.2Посмотрим на зарплаты. У какого количества вакансий заполнено хотя бы одно из двух полей с зарплатой?

In [53]:
query = """
    SELECT
        COUNT(*) as vacancy_count
    FROM
        vacancies
    WHERE
        salary_from IS NOT NULL
        OR salary_to IS NOT NULL;
"""
cursor.execute(query)

# Извлечение результата
vacancy_count = cursor.fetchone()[0]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Количество вакансий с заполненным хотя бы одним полем зарплаты: {vacancy_count}")

Количество вакансий с заполненным хотя бы одним полем зарплаты: 24073


4.3Найдите средние значения для нижней и верхней границы зарплатной вилки. Округлите значения до целого числа.

In [55]:
query = """
    SELECT
        ROUND(AVG(salary_from)) as avg_salary_from,
        ROUND(AVG(salary_to)) as avg_salary_to
    FROM
        vacancies
    WHERE
        salary_from IS NOT NULL
        AND salary_to IS NOT NULL;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()
avg_salary_from = result[0]
avg_salary_to = result[1]

# Закрытие курсора и соединения
cursor.close()
connection.close()

print(f"Средняя нижняя граница зарплатной вилки: {avg_salary_from}")
print(f"Средняя верхняя граница зарплатной вилки: {avg_salary_to}")

Средняя нижняя граница зарплатной вилки: 70484
Средняя верхняя граница зарплатной вилки: 107664


4.4Напишите запрос, который выведет количество вакансий для каждого сочетания типа рабочего графика (schedule) и типа трудоустройства (employment), используемого в вакансиях. Какая пара находится на втором месте по популярности?

In [57]:
query = """
    SELECT
        schedule,
        employment,
        COUNT(*) as vacancy_count
    FROM
        vacancies
    GROUP BY
        schedule, employment
    ORDER BY
        vacancy_count DESC
    LIMIT 2;  -- Для нахождения второй по популярности пары
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результатов
for i, row in enumerate(result):
    schedule, employment, vacancy_count = row
    print(f"Место {i+1}: {schedule}, {employment}, Количество вакансий: {vacancy_count}")

Место 1: Полный день, Полная занятость, Количество вакансий: 35367
Место 2: Удаленная работа, Полная занятость, Количество вакансий: 7802


4.5Напишите запрос, выводящий значения поля Требуемый опыт работы (experience) в порядке возрастания количества вакансий, в которых указан данный вариант опыта.

In [61]:
query = """
    SELECT
        experience,
        COUNT(*) as vacancy_count
    FROM
        vacancies
    GROUP BY
        experience
    ORDER BY
        vacancy_count ASC;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результатов
for row in result:
    experience, vacancy_count = row
    print(f"Требуемый опыт работы: {experience}, Количество вакансий: {vacancy_count}")

Требуемый опыт работы: Более 6 лет, Количество вакансий: 1337
Требуемый опыт работы: Нет опыта, Количество вакансий: 7197
Требуемый опыт работы: От 3 до 6 лет, Количество вакансий: 14511
Требуемый опыт работы: От 1 года до 3 лет, Количество вакансий: 26152


Вывода по заданиям с 4.1 по 4.5:Наибольшее количество вакансий в Москве (5333), Санкт-Петербурге (2851), Минске (2112), Новосибирске (2006) и Алматы (1892).
Москва и Санкт-Петербург доминируют на рынке труда по количеству вакансий, что ожидаемо для крупнейших экономических центров.В 24073 вакансиях указана зарплата, со средней нижней границей 70484 руб. и средней верхней границей 107664 руб., что дает представление о средней вилке зарплат.Полный день и полная занятость (35367 вакансий) являются наиболее популярными типами занятости.
Удаленная работа с полной занятостью (7802 вакансии) также весьма востребована.аибольшее количество вакансий требует опыта от 1 до 3 лет (26152 вакансий).
Меньше всего вакансий для специалистов с опытом более 6 лет (1337 вакансий), что может свидетельствовать о высоком спросе на специалистов начального и среднего уровня.

5.1Напишите запрос, который позволит узнать, какие работодатели находятся на первом и пятом месте по количеству вакансий.



In [63]:
query = """
    WITH ranked_employers AS (
        SELECT
            employer_id,
            COUNT(*) as vacancy_count,
            ROW_NUMBER() OVER (ORDER BY COUNT(*) DESC) as rn
        FROM
            vacancies
        GROUP BY
            employer_id
    )
    SELECT
        e.name,
        re.vacancy_count,
        re.rn
    FROM
        ranked_employers re
    JOIN
        employers e ON re.employer_id = e.id
    WHERE
        re.rn IN (1, 5);
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результатов
for row in result:
    name, vacancy_count, rn = row
    print(f"Место {rn}: Работодатель: {name}, Количество вакансий: {vacancy_count}")

Место 1: Работодатель: Яндекс, Количество вакансий: 1933
Место 5: Работодатель: Газпром нефть, Количество вакансий: 331


5.2Напишите запрос, который для каждого региона выведет количество работодателей и вакансий в нём.

Среди регионов, в которых нет вакансий, найдите тот, в котором наибольшее количество работодателей. Впишите его название в поле ниже в том виде, который вернул запрос.

In [68]:
query = """
   SELECT
    a.name,
    COUNT(DISTINCT e.id) AS employer_count,
    COUNT(v.id) AS vacancy_count
FROM
    areas a
LEFT JOIN
    vacancies v ON a.id = v.area_id
LEFT JOIN
    employers e ON v.employer_id = e.id
GROUP BY
    a.name
HAVING
    COUNT(v.id) = 0
ORDER BY
    employer_count DESC
LIMIT 1;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результатов
for row in result:
    area, employer_count, vacancy_count = row
    print(f"Регион: {area}, Количество работодателей: {employer_count}, Количество вакансий: {vacancy_count}")

Регион: Абатское, Количество работодателей: 0, Количество вакансий: 0


5.3Для каждого работодателя посчитайте количество регионов, в которых он публикует свои вакансии.

Выберите максимальное значение из получившегося списка.

In [79]:
query = """
    SELECT
        MAX(region_count) as max_region_count
    FROM (
        SELECT
            employer_id,
            COUNT(DISTINCT area_id) as region_count
        FROM
            vacancies
        GROUP BY
            employer_id
    ) as subquery;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    max_region_count = result[0]
    print(f"Максимальное количество регионов, в которых работодатель публикует свои вакансии: {max_region_count}")
else:
    print("Не найдено данных")

Максимальное количество регионов, в которых работодатель публикует свои вакансии: 181


5.4Напишите запрос для подсчёта количества работодателей, у которых не указана сфера деятельности.

Введите количество, которое вернул запрос, в поле ниже.

In [77]:
query = """
    SELECT
        COUNT(*) as employer_count
    FROM
        employers e
    LEFT JOIN
        employers_industries ei ON e.id = ei.employer_id
    WHERE
        ei.industry_id IS NULL;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    employer_count = result[0]
    print(f"Количество работодателей без указанной сферы деятельности: {employer_count}")
else:
    print("Не найдено данных")

Количество работодателей без указанной сферы деятельности: 8419


5.5Напишите запрос, чтобы узнать название компании, находящейся на третьем месте в алфавитном списке (по названию) компаний, у которых указано четыре сферы деятельности.

Введите в поле ниже название этой компании так же, как оно указано в результате запроса.

In [84]:
query = """
    SELECT
        e.name AS company_name
    FROM
        employers e
    JOIN
        employers_industries ei ON e.id = ei.employer_id
    GROUP BY
        e.id
    HAVING
        COUNT(ei.industry_id) = 4
    ORDER BY
        e.name
    OFFSET 2 LIMIT 1;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    company_name = result[0]
    print(f"Компания на третьем месте в алфавитном списке компаний с четырьмя сферами деятельности: {company_name}")
else:
    print("Не найдено данных")

Компания на третьем месте в алфавитном списке компаний с четырьмя сферами деятельности: 2ГИС


5.6 С помощью запроса выясните, у какого количества работодателей в качестве сферы деятельности указана «Разработка программного обеспечения».

In [89]:
query = """
    SELECT
        COUNT(DISTINCT e.id) AS software_development_employers
    FROM
        employers e
    JOIN
        employers_industries ei ON e.id = ei.employer_id
    JOIN
        industries i ON ei.industry_id = i.id
    WHERE
        i.name = 'Разработка программного обеспечения';
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    software_development_employers = result[0]
    print(f"Количество работодателей в сфере 'Разработка программного обеспечения': {software_development_employers}")
else:
    print("Не найдено данных")

Количество работодателей в сфере 'Разработка программного обеспечения': 3553


5.7Для компании «Яндекс» выведите список регионов-миллионников , в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавьте строку Total с общим количеством вакансий компании, собранных в этой таблице. Должна получиться выборка такого вида (приведён пример результата для компании SberTech)

In [92]:
query = """
    WITH million_cities AS (
        SELECT id, name
        FROM areas
        WHERE name IN (
            'Москва', 'Санкт-Петербург', 'Новосибирск', 'Екатеринбург', 'Нижний Новгород', 'Казань',
            'Челябинск', 'Омск', 'Самара', 'Ростов-на-Дону', 'Уфа', 'Красноярск', 'Пермь',
            'Воронеж', 'Волгоград'
        )
    )
    SELECT
        a.name AS region,
        COUNT(v.id) AS vacancies_count
    FROM
        vacancies v
    JOIN
        employers e ON v.employer_id = e.id
    JOIN
        million_cities a ON v.area_id = a.id
    WHERE
        e.name = 'Яндекс'
    GROUP BY
        a.name
    ORDER BY
        a.name;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchall()

# Преобразование результата в DataFrame для удобного добавления строки Total
df = pd.DataFrame(result, columns=['region', 'vacancies_count'])

# Подсчет общего количества вакансий и добавление строки Total
total_vacancies = df['vacancies_count'].sum()
df.loc[len(df)] = ['Total', total_vacancies]

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
print(df)

# Печать количества строк в выборке
print(f"Количество строк в выборке: {len(df)}")

             region  vacancies_count
0         Волгоград               24
1           Воронеж               32
2      Екатеринбург               39
3            Казань               25
4        Красноярск               23
5            Москва               54
6   Нижний Новгород               36
7       Новосибирск               35
8              Омск               21
9             Пермь               25
10   Ростов-на-Дону               25
11           Самара               26
12  Санкт-Петербург               42
13              Уфа               26
14        Челябинск               22
15            Total              455
Количество строк в выборке: 16


выводы по заданиям с 5.1 по 5.7:Яндекс (1933 вакансии) и Газпром нефть (331 вакансия) являются лидерами по количеству предложений.8419 работодателей не указали свою сферу деятельности, что затрудняет анализ специфики их предложений.3553 работодателя работают в сфере разработки ПО.Яндекс имеет вакансии в 16 крупных городах, с общим количеством 455 ваканси

6.1Сколько вакансий имеет отношение к данным?

In [94]:
query = """
    SELECT
        COUNT(*) as data_related_vacancies
    FROM
        vacancies
    WHERE
        LOWER(name) LIKE '%data%' OR LOWER(name) LIKE '%данн%';
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    data_related_vacancies = result[0]
    print(f"Количество вакансий, имеющих отношение к данным: {data_related_vacancies}")
else:
    print("Не найдено данных")

Количество вакансий, имеющих отношение к данным: 1771


6.2Сколько есть подходящих вакансий для начинающего дата-сайентиста?

In [96]:
query = """
    SELECT
        COUNT(*) as suitable_vacancies
    FROM
        vacancies
    WHERE
        (LOWER(name) LIKE '%data scientist%' OR
         LOWER(name) LIKE '%data science%' OR
         LOWER(name) LIKE '%исследователь данных%' OR
         (LOWER(name) LIKE '%ml%' AND LOWER(name) NOT LIKE '%html%') OR
         LOWER(name) LIKE '%machine learning%' OR
         LOWER(name) LIKE '%машинн%обучен%') AND
        (LOWER(name) LIKE '%junior%' OR
         experience = 'Нет опыта' OR
         employment = 'Стажировка');
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    suitable_vacancies = result[0]
    print(f"Количество подходящих вакансий для начинающего дата-сайентиста: {suitable_vacancies}")
else:
    print("Не найдено данных")

Количество подходящих вакансий для начинающего дата-сайентиста: 51


6.3Сколько есть вакансий для DS, в которых в качестве ключевого навыка указан SQL или postgres?

В качестве ответа запишите количество, которое вернул запрос.

In [99]:
query = """
    SELECT
        COUNT(*) as ds_sql_postgres_vacancies
    FROM
        vacancies
    WHERE
        (LOWER(name) LIKE '%data scientist%' OR
         LOWER(name) LIKE '%data science%' OR
         LOWER(name) LIKE '%исследователь данных%' OR
         (LOWER(name) LIKE '%ml%' AND LOWER(name) NOT LIKE '%html%') OR
         LOWER(name) LIKE '%machine learning%' OR
         LOWER(name) LIKE '%машинн%обучен%') AND
        (LOWER(key_skills) LIKE '%sql%' OR
         LOWER(key_skills) LIKE '%postgres%');
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    ds_sql_postgres_vacancies = result[0]
    print(f"Количество вакансий для Data Scientist с ключевым навыком SQL или Postgres: {ds_sql_postgres_vacancies}")
else:
    print("Не найдено данных")

Количество вакансий для Data Scientist с ключевым навыком SQL или Postgres: 229


6.4С помощью запроса, аналогичного предыдущему, проверьте, насколько популярен Python в требованиях работодателей к DS. Вычислите количество вакансий, в которых в качестве ключевого навыка указан Python.

В качестве ответа запишите количество, которое вернул запрос.

In [102]:
query = """
    SELECT
        COUNT(*) as ds_python_vacancies
    FROM
        vacancies
    WHERE
        (LOWER(name) LIKE '%data scientist%' OR
         LOWER(name) LIKE '%data science%' OR
         LOWER(name) LIKE '%исследователь данных%' OR
         (LOWER(name) LIKE '%ml%' AND LOWER(name) NOT LIKE '%html%') OR
         LOWER(name) LIKE '%machine learning%' OR
         LOWER(name) LIKE '%машинн%обучен%') AND
        LOWER(key_skills) LIKE '%python%';
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    ds_python_vacancies = result[0]
    print(f"Количество вакансий для Data Scientist с ключевым навыком Python: {ds_python_vacancies}")
else:
    print("Не найдено данных")

Количество вакансий для Data Scientist с ключевым навыком Python: 357


6.5Сколько ключевых навыков в среднем указывают в вакансиях для DS?

In [104]:
query = """
    SELECT
        AVG(LENGTH(key_skills) - LENGTH(REPLACE(key_skills, CHR(9), '')) + 1) as avg_skills
    FROM
        vacancies
    WHERE
        (LOWER(name) LIKE '%data scientist%' OR
         LOWER(name) LIKE '%data science%' OR
         LOWER(name) LIKE '%исследователь данных%' OR
         (LOWER(name) LIKE '%ml%' AND LOWER(name) NOT LIKE '%html%') OR
         LOWER(name) LIKE '%machine learning%' OR
         LOWER(name) LIKE '%машинн%обучен%') AND
         key_skills IS NOT NULL;
"""
cursor.execute(query)

# Извлечение результата
result = cursor.fetchone()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
if result:
    avg_skills = result[0]
    print(f"Среднее количество ключевых навыков в вакансиях для Data Scientist: {avg_skills:.2f}")
else:
    print("Не найдено данных")

Среднее количество ключевых навыков в вакансиях для Data Scientist: 6.55


6.6Напишите запрос, позволяющий вычислить, какую зарплату для DS в среднем указывают для каждого типа требуемого опыта (уникальное значение из поля experience).Выясните, на какую зарплату в среднем может рассчитывать дата-сайентист с опытом работы от 3 до 6 лет. Результат округлите до целого числа.

In [106]:
query = """
    SELECT
        experience,
        ROUND(AVG(COALESCE((salary_from + salary_to) / 2, salary_from, salary_to))) as avg_salary
    FROM
        vacancies
    WHERE
        (LOWER(name) LIKE '%data scientist%' OR
         LOWER(name) LIKE '%data science%' OR
         LOWER(name) LIKE '%исследователь данных%' OR
         (LOWER(name) LIKE '%ml%' AND LOWER(name) NOT LIKE '%html%') OR
         LOWER(name) LIKE '%machine learning%' OR
         LOWER(name) LIKE '%машинн%обучен%') AND
        (salary_from IS NOT NULL OR salary_to IS NOT NULL)
    GROUP BY
        experience;
"""
cursor.execute(query)

# Извлечение результата
results = cursor.fetchall()

# Закрытие курсора и соединения
cursor.close()
connection.close()

# Печать результата
for row in results:
    print(f"Опыт: {row[0]}, Средняя зарплата: {row[1]} руб.")

# Поиск средней зарплаты для опыта от 3 до 6 лет
avg_salary_3_6_years = next((row[1] for row in results if row[0] == 'От 3 до 6 лет'), None)
if avg_salary_3_6_years is not None:
    print(f"Средняя зарплата для Data Scientist с опытом работы от 3 до 6 лет: {avg_salary_3_6_years} руб.")
else:
    print("Не найдено данных для опыта от 3 до 6 лет")

Опыт: Более 6 лет, Средняя зарплата: 157933 руб.
Опыт: От 3 до 6 лет, Средняя зарплата: 256454 руб.
Опыт: Нет опыта, Средняя зарплата: 74643 руб.
Опыт: От 1 года до 3 лет, Средняя зарплата: 144233 руб.
Средняя зарплата для Data Scientist с опытом работы от 3 до 6 лет: 256454 руб.


Выводы по заданиям 6.1 по 6.6:1771 вакансия относится к работе с данными.51 вакансия подходит для начинающего дата-сайентиста.229 вакансий требуют навыков SQL или Postgres.357 вакансий требуют навыков Python.В среднем указывают 6.55 ключевых навыков в вакансиях для Data Scientist.Средняя зарплата для Data Scientist с опытом от 3 до 6 лет составляет 256454 руб., что значительно выше по сравнению с другими категориями опыта.

Общие выводы:На рынке труда преобладают вакансии с полным рабочим днем и полной занятостью.
Удаленная работа также занимает значительное место, что отражает текущие тенденции на рынке труда.
Москва и Санкт-Петербург являются основными центрами трудовой активности с наибольшим количеством вакансий.
Другие крупные города, такие как Минск, Новосибирск и Алматы, также имеют значительное количество предложений.
Большинство вакансий ориентированы на специалистов с опытом от 1 до 3 лет, что может указывать на высокий спрос на профессионалов среднего уровня.
Высокий спрос на специалистов среднего уровня и недостаток вакансий для высококвалифицированных специалистов могут свидетельствовать о нехватке кадров с большим опытом.
Преобладание вакансий в крупных городах указывает на концентрацию экономической активности и рабочих мест в этих регионах.
Значительное количество вакансий в сфере разработки программного обеспечения подтверждает важность IT-сектора на современном рынке труда.
Интерес к навыкам работы с данными, таким как SQL и Python, подчеркивает важность аналитических и технических компетенций в текущих рыночных условиях.
Общий вывод: рынок труда активно развивается, с широким спектром предложений в различных регионах и сферах деятельности. При этом существует значительный спрос на специалистов с опытом от 1 до 3 лет, а также на IT-специалистов и аналитиков данных.